webapp/sources/rudder/rudder-web/src/main/scala/bootstrap/liftweb/UserSessionInvalidationFilter.scala

```diff
@@ -146,13 +149,89 @@ class UserSessionInvalidationFilter(userRepository: UserRepository, userDetailLi
       val userDetails = auth.getPrincipal
       userDetails match {
         case user: RudderUserDetail =>
-          val status = userStatuses.get.map(_.getOrElse(user.getUsername, UserStatus.Deleted))
-          status.runNow match {
-            case UserStatus.Active => None
-            case status            => Some(s"user status changed to ${status.value}")
+          implicit val userDetail: RudderUserDetail = user
+          val cached = userCache.get.map(_.get(user.getUsername)).runNow
+          cached match {
+            case checkUser(reason) => Some(reason)
+            case _                 => None
           }
         case _ => None
       }
     }
   }
 }
+
+private[liftweb] object UserSessionInvalidationFilter {
+
+  /**
+   * Match an user in cache : if no user, it means its cache has changed, with a known or unknown cause of invalidity
+   * if some user, it's ok, unless the last known status is invalid
+   */
+  object checkUser {
+    def unapply(user: Option[Int])(implicit userDetail: RudderUserDetail): Option[String] = {
+      user match {
+        case None                                                   => Some("user is unknown")
+        case Some(HASH_USER_INVALID_STATUS)                         => Some("user status has been updated to an invalid one")
+        // user hash has changed
+        case Some(hash) if hashRudderUserDetail(userDetail) != hash => Some("user access to Rudder has been updated")
```

Comment:
not really because the admin hash itself is stored when cache is reloaded. In fact the case for the admin would be merged with the last one (resulting in `None`), but to mark it as special I kept it.
As discussed, I put right below the status case instead : https://github.com/Normation/rudder/pull/5866/commits/2023ab520b281a3d4251e4b6721a0c45465e9187